### ISSUE TITLE
Build fails, uncaught exception error

### ISSUE THREAD

**Issue by calexil** [lostgoat/gputool#1]
Log of failed build here:

http://dpaste.com/0SYXC91

**Comment by lostgoat** [comment]
Hey,

Thanks for the issue, it seems like the toolchain used doesn't support std::uncaught_exceptions. I should probably add a cmake check for that, or remove the usage.

Do you mind letting me know what environment you are using?

**Comment by calexil** [comment]
sure here you go:

http://dpaste.com/26T28PF

**Comment by calexil** [comment]
hmu on irc for faster responses:

spotchat, freenode, and snoonet

uname: calexil

**Comment by lostgoat** [comment]
Added some extra checks to cmake and a friendlier error message. 

Can I trouble you to now run:
1) "export CXX=different-toolchain-bin"
2) rm -rf build/
3) Run regular build steps

Where different-toolchain points to a new-ish gcc or clang? Usually your distribution may include different versions in the package management system, and after installing them you'll be able to refer to them as e.g. g++-4.7

For example, I tried:
 * export CXX=clang++-3.8 -> works
 * export CXX=g++-6 -> works
 * export CXX=g++-4.7 -> fails
